src/psyclone/tests/psyir/frontend/fparser2_test.py

```diff
@@ -794,6 +795,56 @@ def test_process_declarations():
             "interface" in str(error.value))
 
 
+@pytest.mark.usefixtures("f2008_parser")
+def test_declarations_with_initialisations(fortran_reader, fortran_writer):
+    '''Test that Fparser2Reader keeps all the variable initialisation
+    expressions, even thought some may end up in UnknownTypes for now.
+    '''
+
+    psyir = fortran_reader.psyir_from_source(
+        """
+        module test
+            integer :: a = 1
+            integer, save :: b = 1
+            integer, parameter :: c = 1
+            contains
+            subroutine mysub()
+                integer :: d = 1
+                integer, save :: e = 1
+                integer, parameter :: f = 1
+            end subroutine mysub
+        end module test
+        """)
+
+    inner_st = psyir.walk(Routine)[0].symbol_table
+    # All initialisation variables are DataSymbols
+    assert isinstance(inner_st.lookup('a'), DataSymbol)
+    assert isinstance(inner_st.lookup('b'), DataSymbol)
+    assert isinstance(inner_st.lookup('c'), DataSymbol)
+    assert isinstance(inner_st.lookup('d'), DataSymbol)
+    assert isinstance(inner_st.lookup('e'), DataSymbol)
+    assert isinstance(inner_st.lookup('f'), DataSymbol)
+
+    # When it is not a parameter they are unknown interface and datatype
+    assert isinstance(inner_st.lookup('a').interface, UnknownInterface)
```

Comment:
Can we do better than this - i.e. we know where the declaration is, just not what it contains?

Reply:
It is not that easy at the moment because the _process_decln just returns a NotImplementedError for this. And then what catches the error has to use UnknownInterface because there are other cases that produce the same error.

I am not sure if it is worth refactoring for this case, or we just implement support for initialisations. But in any case it seems for another PR.

Reply:
OK, agreed.